pkg/p2p/libp2p/libp2p.go

```diff
@@ -179,9 +185,13 @@ func New(ctx context.Context, signer beecrypto.Signer, networkID uint64, overlay
 		)
 	}
 
-	if o.PrivateKey != nil {
+	if o.PrivateKey != nil { // TODO fixme
+		myKey, _, err := crypto.ECDSAKeyPairFromKey(o.PrivateKey)
+		if err != nil {
+			return nil, err
+		}
 		opts = append(opts,
-			libp2p.Identity((*crypto.Secp256k1PrivateKey)(o.PrivateKey)),
```

Comment:
`Secp256k1` is no longer supported in later versions of libp2p, it has moved to using `Secp256r1` and `x509`